code/modules/gamemaster/actions/electrified_door.dm

```diff
@@ -0,0 +1,70 @@
+/datum/gm_action/electrified_door
+	name = "airlock short-circuit"
+	departments = list(ROLE_ENGINEERING, ROLE_MEDICAL)
+	chaotic = 10
+	var/obj/machinery/door/airlock/chosen_door
+	var/list/area/excluded = list(
+		/area/shuttle,
+		/area/crew_quarters
+	)
+	var/severity
+
+/datum/gm_action/electrified_door/set_up()
+	var/list/area/grand_list_of_areas = get_station_areas(excluded)
+
+	severity = pickweight(EVENT_LEVEL_MUNDANE = 10,
+		EVENT_LEVEL_MODERATE = 5,
+		EVENT_LEVEL_MAJOR = 1
+		)
+
+	//try 10 times
+	for(var/i in 1 to 10)
+		var/area/A = pick(grand_list_of_areas)
+		var/list/obj/machinery/door/airlock/target_doors = list()
+		for(var/obj/machinery/door/airlock/target_door in A.contents)
+			target_doors += target_door
+		target_doors = shuffle(target_doors)
+
+		for(var/obj/machinery/door/airlock/target_door in target_doors)
+			if(!target_door.isElectrified() && target_door.arePowerSystemsOn() && target_door.maxhealth == target_door.health)
+				chosen_door = target_door
+				return
+
+/datum/gm_action/electrified_door/start()
+	..()
+	if(!chosen_door)
+		return
+	if(prob(33))
+		chosen_door.visible_message("<span class='danger'>\The [chosen_door]'s panel sparks!</span>")
+	chosen_door.set_safeties(0)
+	if(severity >= EVENT_LEVEL_MODERATE)
+		chosen_door.electrify(-1)
```

Comment:
Yeah, I somehow read over this without actually writing something down although I vaguely remember talking about it on TS.  In the interest of stuff not getting lost in Discord, can we please have some sort of permanent SOMETHING visibly wrong with the door so that Joe Blow joining after the radio announcement (... is there even a radio announcement?) and not seeing the sparks doesn't get randomly screwed because we decided to run meteor shields?  Thanks.

((Actually, that's a good question - how do you know where to start looking if you don't see the sparks?  Not entirely sure how notifications work for these))